workflow/artifacts/s3/s3.go

```diff
@@ -180,7 +180,27 @@ func (s3Driver *ArtifactDriver) Delete(artifact *wfv1.Artifact) error {
 		if err != nil {
 			return err
 		}
-		return s3cli.Delete(artifact.S3.Bucket, artifact.S3.Key)
+
+		isDir, err := s3cli.IsDirectory(artifact.S3.Bucket, artifact.S3.Key)
+		if err != nil {
+			return fmt.Errorf("failed to test if %s is a directory: %v", artifact.S3.Key, err)
```

Comment:
```suggestion
			return fmt.Errorf("failed checking if %s is a directory: %v", artifact.S3.Key, err)
```

the word "test" has a connotation around testing, so would be better to avoid that. I was confused by the error message myself when I first read it.